src/firmware/app/world/firmware_robot.c

```diff
@@ -226,12 +226,24 @@ void velocity_wheels_setLocalVelocity(
     float linear_velocity_y = control_msg.velocity.y_component_meters;
     float angular_velocity  = control_msg.angular_velocity.radians_per_second;
 
-    float robot_velocity[3];
-    robot_velocity[0] = linear_velocity_x;
-    robot_velocity[1] = linear_velocity_y;
-    robot_velocity[2] = angular_velocity;
+    float global_robot_velocity[2];
+    global_robot_velocity[0] = linear_velocity_x;
+    global_robot_velocity[1] = linear_velocity_y;
```

Comment:
shouldn't linear_velocity_x and linear_velocity_y already be in local velocities since the function is called velocity_wheels_setLocalVelocity?

Reply:
I could move this conversion from global to local velocity to where it's called instead. Is `TbotsProto_DirectControlPrimitive_DirectVelocityControl control_msg` supposed to contain global or local velocity?

Reply:
it's supposed to be local velocities

Reply:
Fixed